src/main/java/net/sacredlabyrinth/phaed/simpleclans/managers/ChatManager.java

```diff
@@ -54,57 +54,43 @@ public ChatManager(SimpleClans plugin) {
     @Subscribe
     public void setupDiscord(DiscordReadyEvent event) {
         guild = DiscordSRV.getPlugin().getMainGuild();
+        List<String> clanTags = getClanTags();
         textCategories = settingsManager.getStringList(DISCORDCHAT_TEXT_CATEGORY_IDS).stream().
                 filter(this::categoryExists).collect(Collectors.toList());
 
-        List<String> clanTags = plugin.getClanManager().getClans().stream().
-                filter(Clan::isVerified).
-                filter(clan -> !getDiscordPlayersId(clan.getTag()).isEmpty() || clan.isPermanent()).
-                map(Clan::getTag).
-                collect(Collectors.toList());
-
-        List<String> channels = guild.getTextChannels().stream().
-                map(GuildChannel::getName).
-                distinct().
-                collect(Collectors.toList());
-
-        clanTags.removeAll(channels);
-
-        if (textCategories.isEmpty()) {
+        int categoriesToCreate = ((int) Math.ceil((double) clanTags.size() / MAX_CHANNELS_PER_CATEGORY)) - textCategories.size();
+        for (int i = 0; i < categoriesToCreate; i++) {
             createCategory();
         }
 
         for (String textCategory : textCategories) {
-            Category category = getCategory(textCategory);
-            for (String clanTag : clanTags) {
-                createChannel(category, clanTag);
+            Category category = Objects.requireNonNull(guild.getCategoryById(textCategory),
+                    "Category " + textCategory + " can't be created!");
+            Iterator<String> tagIter = clanTags.iterator();
+            while (tagIter.hasNext()) {
+                if (createChannel(category, tagIter.next())) {
+                    tagIter.remove();
+                } else {
+                    break;
+                }
```

Comment:
Fixes `ConcurrentModificationException` really? 
`createChannel` doesn't make a new category now.

Does this lines makes sense then? 🤔